test/TesterInternal/StreamingTests/StreamReliabilityTests.cs

```diff
@@ -887,16 +896,21 @@ private void RestartAllSilos()
             output.WriteLine("\n\n\n\n-----------------------------------------------------\n" +
                             "Restarting all silos - Old Primary={0} Secondary={1}" +
                             "\n-----------------------------------------------------\n\n\n",
-                            this.HostedCluster.Primary.SiloAddress, this.HostedCluster.Secondary.SiloAddress);
+                            this.HostedCluster.Primary.SiloAddress, this.HostedCluster.SecondarySilos.First().SiloAddress);
+
+            //this.HostedCluster.RestartDefaultSilos();
+            foreach (var silo in this.HostedCluster.GetActiveSilos())
```

Comment:
GetActiveSilos().ToList() otherwise the collection is modified as you are enumerating. I'll add it here and push to your branch